include/gz/sim/components/EntityWrench.hh

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef GZ_SIM_COMPONENTS_ENTITYWRENCH_HH_
+#define GZ_SIM_COMPONENTS_ENTITYWRENCH_HH_
+
+#include <string>
+#include <unordered_map>
+
+#include <gz/msgs/entity_wrench.pb.h>
+#include <gz/sim/components/Component.hh>
+#include <gz/sim/components/Factory.hh>
+#include <gz/sim/components/Serialization.hh>
+#include <gz/sim/config.hh>
+
+namespace gz
+{
+namespace sim
+{
+// Inline bracket to help doxygen filtering.
+inline namespace GZ_SIM_VERSION_NAMESPACE {
+namespace serializers
```

Comment:
Yes - I'm trying that now, but using a `map<string, EntityWrenches>` element instead of a `repeated EntityWrenches` in the protobuf. I'll place the additional method side by side with the others for comparison.

I'm wondering whether there is some hashing of the component payload that I'm missing which is causing the update to be skipped? It's quite difficult navigating around the ECM inside the debugger - typelists and so on are quite challenging to step through! It would be nice to have an ECM debug tool to help trace down what's going on.